uint32 vector_len
object[] object_vector